src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -1014,9 +1015,18 @@ private async Task ShowManageLibraryPackageForSolutionDialogAsync(string searchT
                     // event handler is set up here because the PMUI can only be shown once the solution has fully loaded in as it is not feasible to open the PMUI when
                     // there is no solution associated with it
                     EventHandler handler = null;
+
+                    SolutionPickerViewModel viewModel = new SolutionPickerViewModel();
+
+                    var storageManagerFactory = new CodeContainerStorageManagerFactory(this, JoinableTaskFactory);
+                    await viewModel.PopulateSolutionListAsync(storageManagerFactory);
+
+                    SolutionPickerWindow window = new SolutionPickerWindow(viewModel);
+
                     handler = (s, e) =>
                     {
                         SolutionManager.Value.SolutionOpened -= handler;
+                        window.DialogResult = true;
```

Comment:
It should be the responsibility of the dialog to set its DialogResult, not the caller.